packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -132,6 +133,43 @@ export type GenerateSummaryResult =
     | IUploadSummaryResult
     | ISubmitSummaryOpResult;
 
+export interface IBroadcastSummaryResult {
+    readonly summarizeOp: ISummaryOpMessage;
+    readonly broadcastDuration: number;
+}
+
+export interface IAckNackSummaryResult {
+    readonly summaryAckNackOp: ISummaryAckMessage | ISummaryNackMessage;
+    readonly ackNackDuration: number;
+}
+
+export type SummarizeResultPart<T> = {
+    success: true;
+    data: T;
+} | {
+    success: false;
+    data: T | undefined;
+    message: string;
+    error: any;
+};
+
+export interface ISummarizeResult {
+    /** Resolves when we generate, upload, and submit the summary */
+    readonly generateSummary: Promise<SummarizeResultPart<GenerateSummaryResult>>;
+    /** Resolves when we see our summarize op broadcast; is sequence number of op */
+    readonly broadcastSummaryOp: Promise<SummarizeResultPart<IBroadcastSummaryResult>>;
```

Comment:
nit: Maybe this should be called `broadcastSummaryResult` or just `broadcastSummary` as it returns more than the op.

Reply:
So the name isn't about what it returns, but about which stage in the process it represents.
So the first stage is to generate the summary. The second stage is to wait for the summary op to be broadcast. And the third stage is to wait for the summary ack to be received.

That being said, I also am not a huge fan of the naming here.. I'll rethink the names.

Reply:
I changed it. Let me know what you think.

Reply:
Looks much better.